app/scripts/modules/core/application/modal/validation/validateApplicationName.directive.ts

```diff
@@ -33,24 +32,27 @@ class ValidateApplicationNameController implements ng.IComponentController {
   }
 }
 
+
+@DirectiveFactory('applicationNameValidator')
```

Comment:
So does this mean the parameter name on line 26/25 above has to match the string value of this name?  If so, I'd like to investigate https://github.com/ngParty/ng-metadata before hand-rolling our own solutions because at least then we'd be able to `@Name` our parameters.  That project may not work for us, but I'd like to see.  It's been on my todo list....

Thoughts?